### PR TITLE
CLIENTS-799: Unify modals templates, styles, and JS

### PR DESCRIPTION
**Pull request by tomasstark**
This PR addresses [CLIENTS-799](https://laterpay.atlassian.net/browse/CLIENTS-799).

- This is mostly a cleanup of code handling modals. For some modals, template and CSS was unnecessarily tied to specific modal instance and could be abstracted more which is what this PR addresses.

- It updates templates of all modals to use new unified markup.

- It cleans up unused CSS created per-modal previously.

- All modals in the plugin now utilize our minimal [RevGenModal](https://github.com/laterpay/revenue-generator/blob/develop/assets/src/js/utils/rev-gen-modal.js) module.

### CODE REVIEW

File: assets/src/js/pages/revenue-generator-contribution.js [hunk 1]
```diff
@@ -861,25 +860,168 @@ import { __, sprintf } from '@wordpress/i18n';
...
 			};
 
+			const showAccountActivationModal = function() {
+				new RevGenModal( {
+					id: 'rg-modal-account-activation',
+					keepOpen: true,
+					templateData: {},
+					onConfirm: async () => {
+						showAccountModal();
+					},
```
Comment (tomislavhren) [inline assets/src/js/pages/revenue-generator-contribution.js:870]: Isn't async unnecessary you don't await it?

File: assets/src/js/pages/revenue-generator-contribution.js [hunk 2]
```diff
@@ -861,25 +860,168 @@ import { __, sprintf } from '@wordpress/i18n';
...
+					onCancel: async ( e, el ) => {
```
Comment (tomislavhren) [inline assets/src/js/pages/revenue-generator-contribution.js:871]: Same as ⬆️

File: assets/src/js/pages/revenue-generator-contribution.js [hunk 3]
```diff
@@ -861,25 +860,168 @@ import { __, sprintf } from '@wordpress/i18n';
...
+			};
+
+			const showAccountModal = function() {
+				new RevGenModal( {
+					id: 'rg-modal-connect-account',
+					keepOpen: true,
+					templateData: {},
+					onConfirm: async ( e, el ) => {
+						const closeEvent = new Event( 'rev-gen-modal-close' );
+						const $el = $( el );
```
Comment (tomislavhren) [inline assets/src/js/pages/revenue-generator-contribution.js:914]: Very brave with those whitespaces 😂

**Review by tomislavhren — APPROVED** [review]

File: assets/src/js/pages/revenue-generator-contribution.js (hunk 1 above)
Comment (tomasstark) [reply assets/src/js/pages/revenue-generator-contribution.js:870]: You're right, completely unnecessary. Thanks! 🙂

**Pull request merged**